digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/AwsService.scala

```diff
@@ -0,0 +1,299 @@
+package fi.liikennevirasto.digiroad2.service
+
+import fi.liikennevirasto.digiroad2.AssetService
+import fi.liikennevirasto.digiroad2.asset._
+import fi.liikennevirasto.digiroad2.dao.AwsDao
+import fi.liikennevirasto.digiroad2.linearasset.{DynamicAssetValue, DynamicValue, NewLinearAsset, NumericValue, RoadLink, Value}
+import fi.liikennevirasto.digiroad2.service.linearasset._
+import fi.liikennevirasto.digiroad2.service.pointasset.{IncomingObstacle, ObstacleService, PavedRoadService, PedestrianCrossingService, RailwayCrossingService, TrafficLightService, HeightLimit => _, WidthLimit => _}
+import org.json4s._
+import org.json4s.jackson.Serialization
+import org.json4s.jackson.Serialization.write
+import fi.liikennevirasto.digiroad2.client.vvh.VVHClient
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+
+case class Dataset(datasetId: String, geoJson: Map[String, Any], roadlinks: List[List[BigInt]])
+
+class AwsService(vvhClient: VVHClient,
+                 val onOffLinearAssetService: OnOffLinearAssetService,
+                 val roadLinkService: RoadLinkService,
+                 val linearAssetService: LinearAssetService,
+                 val speedLimitService: SpeedLimitService,
+                 val pavedRoadService: PavedRoadService,
+                 val roadWidthService: RoadWidthService,
+                 val manoeuvreService: ManoeuvreService,
+                 val assetService: AssetService,
+                 val obstacleService: ObstacleService,
+                 val pedestrianCrossingService: PedestrianCrossingService,
+                 val railwayCrossingService: RailwayCrossingService,
+                 val trafficLightService: TrafficLightService,
+                 val massTransitLaneService: MassTransitLaneService,
+                 val numberOfLanesService: NumberOfLanesService
+                ) {
+
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  def awsDao: AwsDao = new AwsDao
+
+  implicit val formats = Serialization.formats(NoTypeHints)
+
+  final val AwsUser = "AwsUpdater"
+
+  val allDatasetStatus = Map(
+    0 -> "Inserted successfuly",
+    1 -> "Amount of features and roadlinks do not match",
+    2 -> "Errors in the features",
+    3 -> "Processed successfuly",
+    4 -> "Errors while processing"
+  )
+
+  val allFeatureStatus = Map(
+    0 -> "Inserted successfuly",
+    1 -> "Errors while validating",
+    2 -> "Processed successfuly",
+    3 -> "SpeedLimit with invalid speed",
+    4 -> "PavementClass with invalid pavement class",
+    5 -> "Obstacle with invalid class",
+    6 -> "Invalid sideCode"
+  )
+
+  private def linkIdValidation(linkIds: Set[Long]): Unit = {
+    val roadLinks = roadLinkService.getRoadsLinksFromVVH(linkIds)
+    linkIds.foreach { linkId =>
+      roadLinks.find(road => road.linkId == linkId && road.administrativeClass != State).get
+    }
+  }
```

Comment:
Avoid exception.

You can use something like this:

roadLinks.find(road => road.linkId == linkId && road.administrativeClass != State) match
{
case None => //update feature status with the specific error
}